ui/round/src/view/button.ts

```diff
@@ -90,26 +95,30 @@ function rematchButtons(ctrl: RoundController): MaybeVNodes {
 
 export function standard(
   ctrl: RoundController,
-  condition: ((d: RoundData) => boolean) | undefined,
+  condition: ((d: RoundData) => ButtonState) | undefined,
   icon: string,
   hint: string,
   socketMsg: string,
   onclick?: () => void
 ): VNode {
   // disabled if condition callback is provided and is falsy
-  const enabled = () => !condition || condition(ctrl.data);
+  const enabled = () => !condition || condition(ctrl.data).enabled;
+  const hintFn = () => {
+    if (!condition) return hint;
+    return condition(ctrl.data)?.overrideHint || hint;
```

Comment:
You can actually do `condition.?(..)` here to avoid the if.

Reply:
Good idea, fixed!